typescript/sdk/src/core/HyperlaneCoreDeployer.ts

```diff
@@ -111,14 +131,36 @@ export class HyperlaneCoreDeployer extends HyperlaneDeployer<
       mailbox.address,
     );
 
-    const contracts = {
-      validatorAnnounce,
-      proxyAdmin,
+    let timelockController: TimelockController;
+    if (config.upgradeTimelockDelay) {
+      timelockController = await this.deployTimelock(
+        chain,
+        config.upgradeTimelockDelay,
+        config.owner,
+      );
+      await this.transferOwnershipOfContracts(
+        chain,
+        timelockController.address,
+        { proxyAdmin },
+      );
+      await this.transferOwnershipOfContracts(chain, config.owner, { mailbox });
+    } else {
+      // mock this for consistent serialization
+      timelockController = TimelockController__factory.connect(
```

Comment:
Is this really necessary? Shouldn't it just be optional?

Reply:
this allows us to avoid @ts-ignore